pkg/volume/csi/csi_block.go

```diff
@@ -252,22 +247,66 @@ func (m *csiBlockMapper) SetUpDevice() (string, error) {
 	}
 
 	// Call NodeStageVolume
-	stagingPath, err := m.stageVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment)
+	_, err = m.stageVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment)
 	if err != nil {
 		return "", err
 	}
 
-	// Call NodePublishVolume
-	publishPath, err := m.publishVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment, stagingPath)
+	return m.getPublishPath(), nil
```

Comment:
This is a little bit confusing. We NodeStage the volume to staging path but return publish path, which is currently empty, with the expectation that MapPodDevice will later NodePublish the volume there.

Reply:
TBH I think it's fine, I don't want to refactor this again, but would appreciate some comments for future readers.

Reply:
Thank you for pointing it out. I will add comments, here.